src/transformers/models/flava/configuration_flava.py

```diff
@@ -0,0 +1,611 @@
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        ce_ignore_index (`int`, *optional*, defaults to -100):
+            Cross entropy index to ignore.
+        mim_weight (`float`, *optional*, defaults to 1.0):
+            Weight to be assigned to MIM unimodal loss
+        mlm_weight (`float`, *optional*, defaults to 1.0):
+            Weight to be assigned to MLM unimodal loss
+        global_contrastive_weight (`float`, *optional*, defaults to 1.0):
+            Weight to be assigned to global contrastive cross-alignment loss.
+        itm_weight (`float`, *optional*, defaults to 1.0):
+            Weight to be assigned to image-text matching multimodal loss.
+        mmm_image_weight (`float`, *optional*, defaults to 1.0):
+            Weight to be assigned to MMM loss's image part.
+        mmm_text_weight (`float`, *optional*, defaults to 1.0):
+            Weight to be assigned to MMM loss's text part.
+        global_backprop_contrastive (`bool`, *optional*, defaults to True):
+            Whether to use global backpropgation through all workers in contrastive loss.
+        skip_unmasked_multimodal_encoder (`bool`, *optional*, defaults to True):
+            Whether to skip running unmasked multimodal encoder whose outputs are not used by FLAVA losses.
+        return_loss (`bool`, *optional*, defaults to True):
+            Whether to return loss or not
+
+        kwargs (*optional*):
+            Dictionary of keyword arguments.
+
+    Example:
+
+    ```python
+    >>> from transformers import FLAVAModel, FLAVAForPreTraining, FLAVAConfig
+
+    >>> # Initializing a FLAVAConfig with style configuration
+    >>> configuration = FLAVAConfig()
+
+    >>> # Initializing a FLAVAModel and FLAVAForPreTraining model from the style configuration
+    >>> model = FLAVAModel(configuration)
+    >>> model_pre = FLAVAForPreTraining(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    >>> configuration_pre = model_pre.config
+    ```
+    """
+
+    model_type = "flava"
+    is_composition = True
+
+    def __init__(
+        self,
+        image_config_dict=None,
+        text_config_dict=None,
+        multimodal_config_dict=None,
+        hidden_size=768,
+        layer_norm_eps=1e-12,
+        projection_dim=768,
+        logit_scale_init_value=2.6592,
+        initializer_range=0.02,
+        ce_ignore_index=-100,
+        mim_weight=1.0,
+        mlm_weight=1.0,
+        global_contrastive_weight=1.0,
+        itm_weight=1.0,
+        mmm_image_weight=1.0,
+        mmm_text_weight=1.0,
+        global_backprop_contrastive=True,
+        skip_unmasked_multimodal_encoder=True,
+        return_loss=True,
+        **kwargs
+    ):
+        super().__init__(
+            text_config_dict=text_config_dict,
+            image_config_dict=image_config_dict,
+            multimodal_config_dict=multimodal_config_dict,
```

Comment:
Those attributes are set afterwards. Why pass them here?

Reply:
This actually followed CLIP: https://github.com/huggingface/transformers/blob/db7d6a80e82d66127b2a44b6e3382969fdc8b207/src/transformers/models/clip/configuration_clip.py#L241

But, you are right. These should be assigned afterwards once they have their correct values assigned to them.